Source/JavaScriptCore/wasm/WasmOperations.cpp

```diff
@@ -380,7 +380,7 @@ JSC_DEFINE_JIT_OPERATION(operationWasmToJSExitMarshalArguments, EncodedJSValue,
                 else
                     *access.operator()<uint64_t>(calleeFrame, dst) = raw;
             } else {
-                auto raw = *access.operator()<uint64_t>(argumentRegisters, wasmParam.jsr().payloadGPR() * sizeof(UCPURegister));
+                auto raw = *access.operator()<uint64_t>(argumentRegisters, gprToIndex(wasmParam.jsr().payloadGPR()) * sizeof(UCPURegister));
```

Comment:
Hmm, `gprToIndex` seems inefficient for a constant sized pool of registers... anyway that's a bug for another day.

Reply:
Hmm, so looking at GPRInfo, it seems like the issue is that e.g. `argumentGPR0` _is_ `regT0` on ARM, whereas there's no such correspondence on X86. But if all we're really looking for is `argNum`, then we may not need to do any calculation at all; lemme see how EWS looks when I do it that way.

Reply:
Yeah, that's probably why this worked on ARM64 but not x86.

Reply:
We should really just have the inverse of `    
    static constexpr GPRReg toArgumentRegister(unsigned index)` for each ISA.

Reply:
Oops, using `argNum` directly doesn't work when we have GPRs and FPRs interleaved, of course. 🤦
I like your `toArgumentRegister` idea; happy to do that as a follow-up.